src/main/java/fr/adrienbrault/idea/symfony2plugin/config/yaml/YamlReferenceContributor.java

```diff
@@ -37,5 +39,245 @@ public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar)
+                ),
+            new YAMLScalarServiceReferenceProvider()
+        );
+
+        // services:
+        //     app.service.foo: '@app.service.bar'
+        registrar.registerReferenceProvider(
+            PlatformPatterns
+                .psiElement(YAMLScalar.class)
+                .withParent(
+                    PlatformPatterns
+                        .psiElement(YAMLKeyValue.class)
+                        .withParent(
+                            PlatformPatterns
+                                .psiElement(YAMLMapping.class)
+                                .withParent(
+                                    PlatformPatterns
+                                        .psiElement(YAMLKeyValue.class)
+                                        .withName("services")
+                                )
+                        )
+                ),
+            new YAMLScalarServiceReferenceProvider()
+        );
+
+        // services:
+        //     app.service.foo:
+        //         calls:
+        //             - setBar: [ '@app.service.bar' ]
+        registrar.registerReferenceProvider(
+            PlatformPatterns
+                .psiElement(YAMLScalar.class)
+                .withParent(
+                    PlatformPatterns
+                        .psiElement(YAMLSequenceItem.class)
+                        .withParent(
+                            PlatformPatterns
+                                .psiElement(YAMLSequence.class)
+                                .withParent(
+                                    PlatformPatterns
+                                        .psiElement(YAMLKeyValue.class)
+                                        .withParent(
+                                            PlatformPatterns
+                                                .psiElement(YAMLMapping.class)
+                                                .withParent(
+                                                    PlatformPatterns
+                                                        .psiElement(YAMLSequenceItem.class)
+                                                        .withParent(
+                                                            PlatformPatterns
+                                                                .psiElement(YAMLSequence.class)
+                                                                .withParent(
+                                                                    PlatformPatterns
+                                                                        .psiElement(YAMLKeyValue.class)
+                                                                        .withName("calls")
+                                                                )
+                                                        )
+                                                )
+                                        )
+
+                                )
+                        )
+                ),
```

Comment:
Any suggestion how this could be simplified will be more the welcome :thinking:

Reply:
as long as its tested its fine, thats how its nested :)

i also have no idea to simplify this.